org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/jvmmodel/JvmModelAssociator.java

```diff
@@ -467,6 +467,9 @@ public void removeAssociation(EObject sourceElement, EObject jvmElement) {
 		if (sources != null && sources.remove(sourceElement)) {
```

Comment:
Please make a local var for the `targetToSourceMap(resource)` and use that one here and below.